daemon/src/model/cfd.rs

```diff
@@ -949,22 +963,32 @@ impl Cfd {
     }
 
     /// Given an attestation, find and decrypt the relevant CET.
-    pub fn decrypt_cet(self, attestation: &oracle::Attestation) -> Result<Option<Event>> {
-        anyhow::ensure!(!self.is_final());
+    pub fn decrypt_cet(
+        self,
+        attestation: &oracle::Attestation,
+    ) -> Result<Option<Event>, DecryptCetError> {
+        if self.is_final() {
+            return Err(DecryptCetError::CfdIsFinal);
```

Comment:
WDYT of just returning `Ok(None)` here and putting a `trace` statement in that says "Cannot decrypt CET because CFD is already final."?

Reply:
Changing this fucntion to return `Result<Event, DecryptCetError>`

I think the question here is do we want to handle tracing inside this function or outside. I am leaning to returning concrete errors as much as possible and handling errors higher up.